### ISSUE TITLE
MacOS not saving 0.0.0.0 between hardware restarts

### ISSUE THREAD

**Issue by gwthompson** [ollama/ollama#3639]
### What is the issue?

When I set launchctl setenv OLLAMA_HOST "0.0.0.0" and restart the Ollama ap everything works as expected and I can access the API from other devices on my network. 

However when I reboot my Mac the OLLAMA_HOST reverts back to 127.0.0.1 and I have to run launchctl setenv OLLAMA_HOST "0.0.0.0" again.

### What did you expect to see?

When I run launchctl setenv OLLAMA_HOST "0.0.0.0" and restart my Mac I expected the OLLAMA_HOST to still be set at 0.0.0.0

### Steps to reproduce

1. Run  launchctl setenv OLLAMA_HOST "0.0.0.0"
2. Restart Ollama App
3. Check the logs to make sure it says "Listening on [::]:11434 (version 0.1.31)"
4. Reboot the Mac
5. Check the logs again and it now says "Listening on 127.0.0.1:11434 (version 0.1.31)"

### OS

macOS

### Architecture

arm64

### Ollama version

0.1.31

### GPU

Apple

### CPU

Apple

**Comment by arsaboo** [comment]
That is correct...`OLLAMA_HOST` needs to be set after every boot. It is the expected behavior.

**Comment by dahjson** [comment]
I think we could improve the user experience by including some user configuration settings directly within the Ollama app for connecting to websites or Chrome extensions. That way upon a restart these configurations can be reapplied automatically.

https://github.com/ollama/ollama/issues/3598

**Comment by gwthompson** [comment]
In my particular scenario I am running the Ollama app on a standalone Mac mini and then accessing the API from other devices on both my internal network and my Tailscale network so ideally if the mini reboots for any reason it would have the 0.0.0.0 set properly.

I'm going to look into whether using a cron job to set the OLLAMA_HOST to 0.0.0.0 on reboot would solve my particular issue.

By the way OLLAMA is the best thing I've used in my LLM journey! Great work!

**Comment by lin16303** [comment]
With a script open in the Script Editor app on your Mac, choose File > Export. Enter a name for the app. To save your app in a different location than shown, click the disclosure triangle, then select a folder. Click the File Format pop-up menu, then choose Application. Put the Application in the Open at login and get rid of Ollama

do shell script "launchctl setenv OLLAMA_HOST 0.0.0.0"
do shell script "OLLAMA_NUM_PARALLEL=2"
do shell script "OLLAMA_MAX_LOADED_MODELS=2"
do shell script "OLLAMA_KEEP_ALIVE=24h"
do shell script "open /Applications/ollama.app"

**Comment by lizhen1412** [comment]
Is there a configuration file that can be changed to specify 0.0.0.0 directly?

**Comment by tkoenig89** [comment]
For anyone coming here (like i did) because other settings where also reset after a reboot. I tried @lin16303 approach, which only worked half, so i modified it to have ollama be startet with keepAlive etc. configured:

```bash
do shell script "launchctl setenv OLLAMA_NUM_PARALLEL 2"
do shell script "launchctl setenv OLLAMA_MAX_LOADED_MODELS 2"
do shell script "launchctl setenv OLLAMA_KEEP_ALIVE 24h"
do shell script "open /Applications/ollama.app"
```

Setting the variables without launchctl did not work for me.

**Comment by lizhen1412** [comment]
> For anyone coming here (like i did) because other settings were also reset after a reboot. I tried @lin16303 approach, which only worked half, so i modified it to have ollama be startet with keepAlive etc. configured:
> 
> ```shell
> do shell script "launchctl setenv OLLAMA_NUM_PARALLEL 2"
> do shell script "launchctl setenv OLLAMA_MAX_LOADED_MODELS 2"
> do shell script "launchctl setenv OLLAMA_KEEP_ALIVE 24h"
> do shell script "open /Applications/ollama.app"
> ```
> 
> Setting the variables without launchctl did not work for me.

Have you tried reinstalling it? In the new version, it seems that the OLLAMA_HOST variable can be configured only with sudo startup.

**Comment by lin16303** [comment]
[ollama.app.zip](https://github.com/user-attachments/files/17271934/ollama.app.zip)


> > For anyone coming here (like i did) because other settings were also reset after a reboot. I tried @lin16303 approach, which only worked half, so i modified it to have ollama be startet with keepAlive etc. configured:
> > ```shell
> > do shell script "launchctl setenv OLLAMA_NUM_PARALLEL 2"
> > do shell script "launchctl setenv OLLAMA_MAX_LOADED_MODELS 2"
> > do shell script "launchctl setenv OLLAMA_KEEP_ALIVE 24h"
> > do shell script "open /Applications/ollama.app"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Setting the variables without launchctl did not work for me.
> 
> Have you tried reinstalling it? In the new version, it seems that the OLLAMA_HOST variable can be configured only with sudo startup.

[ollama.app.zip](https://github.com/user-attachments/files/17271937/ollama.app.zip)
work with 0.3.12

**Comment by jfbloom22** [comment]
this worked for me.  Any idea how to set the default context-length?

**Comment by lizhen1412** [comment]
> this worked for me. Any idea how to set the default context-length?

It should be in the source code.

**Comment by jcastro** [comment]
It's quite strange that we need to use a script to set this environment. Shouldn't there be a config file or something the app can read when starting?

**Comment by dhiltgen** [comment]
I think we can close this out now that 0.9.4 has a settings UI to cover this capability which persists.

<img width="650" alt="Image" src="https://github.com/user-attachments/assets/6656d225-f20d-404d-ae61-1b504f358977" />
